src/utils.ts

```diff
@@ -137,9 +138,30 @@ export const recursivelyDownloadTx = async (blockId: string, txIds: string[] = [
       !data.has(parent)
   });
 
-  return recursivelyDownloadTx(blockId, [...txIds, ...newParents], data.set(parsedTx.txId, parsedTx));
+  const newData = data.set(parsedTx.txId, cleanInvalidOutputs(parsedTx));
+
+  return recursivelyDownloadTx(blockId, [...txIds, ...newParents], newData);
 };
 
+/**
+ * Removes invalid tx outputs from the received tx
+ *
+ * @param tx - `FullTx` to remove the invalid outputs
+ */
+export const cleanInvalidOutputs = (tx: FullTx) => ({
+  ...tx,
+  // Filter outputs that we can't handle (script was unable to be decoded)
+  outputs: tx.outputs.filter((output) => {
+    const validDecoded = !isNil(get(output, 'decoded.type'));
+
+    if (!validDecoded) {
+      logger.warn(`Ignoring tx output from tx ${tx.txId} as script couldn't be decoded.`);
```

Comment:
Important to log the output index as well.

Reply:
Fixed in 65d9a5a613fe35fc73c0c1fee05f83f74c30b324